database/database.go

```diff
@@ -272,3 +316,52 @@ func (db DatastoreBacked) DeleteIneligibleDomainStates(domains []string) (err er
 	}
 	return delete(keys)
 }
+
+// GetAllIneligibleDomainStates returns all the ineligible domains in the database
+func (db DatastoreBacked) GetAllIneligibleDomainStates() (states []IneligibleDomainState, err error) {
+	// Set up the datastore context.
+	c, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	client, datastoreErr := db.backend.NewClient(c, db.projectID)
+	if datastoreErr != nil {
+		return states, datastoreErr
+	}
+
+	keys, err := client.GetAll(c, datastore.NewQuery("IneligibleDomainState"), &states)
+	if err != nil {
+		return states, err
+	}
+
+	for i, key := range keys {
+		state := states[i]
+		state.Name = key.Name
+		states[i] = state
+	}
+
+	return states, nil
+}
+
+// SetPendingAutomatedRemoval sets the status of a list of domains to StatusPendingAutoamtedRemoval
+func SetPendingAutomatedRemoval(db Database, domains []string, logf func(fomat string, args ...interface{})) error {
+	setDomainStates := func(domainStates []DomainState) error {
+		logf("Updating %d entries...", len(domainStates))
+
+		if err := db.PutStates(domainStates, blackholeLogf); err != nil {
```

Comment:
Since your function receives a logf parameter, I think you want to pass that in here instead of unconditionally using blackholeLogF